runtime-sdk/src/modules/core/mod.rs

```diff
@@ -133,6 +133,19 @@ pub struct TxSimulationFailure {
     code: u32,
 }
 
+impl TxSimulationFailure {
+    /// Returns true if the failure is "core::Error::OutOfGas" or "evm::OutOfGas".
+    pub fn is_error_out_of_gas(&self) -> bool {
+        (self.module_name == MODULE_NAME && self.code == 12)
+            || (self.module_name == "evm" && self.code == 2 && self.message.contains("out of gas"))
+    }
+
+    /// Returns true if the failure is "evm::Reverted".
+    pub fn is_error_evm_reverted(&self) -> bool {
+        self.module_name == "evm" && self.code == 8
```

Comment:
Hm this doesn't seem ok as this now relies on implementation details of the `evm` module. Can we do this differently?

Is it possible to detect (in the `evm` module) that the revert was due to an out of gas condition? If yes, then we should remap to `core::Error::OutOfGas`, at least during simulation.

Reply:
I think not, it looks like the engine fails transactions that don't have enough of gas for EIP-150 with a revert instead of an "out-of-gas" (after increasing the gas used, the transaction passes). This might be a bug in the upstream evm crate code, will take a look.

Reply:
Fixed, by doing the same as frontier (doing one simulation at max gas and that way knowing if the transaction is failing due to gas limit or something else)